src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -8,77 +8,70 @@ namespace Microsoft.AspNetCore.Razor.TagHelpers
         /// </summary>
-        public TagHelperAttribute()
+        /// <param name="name">The <see cref="Name"/> of the attribute.</param>
+        public TagHelperAttribute(string name)
+            : this(name, value: null, minimized: true)
         {
         }
 
         /// <summary>
-        /// Instantiates a new instance of <see cref="TagHelperAttribute" /> with values provided by the given
-        /// <paramref name="attribute"/>.
+        /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>
+        /// and <paramref name="value"/>. <see cref="Minimized"/> is set to <c>false</c>.
         /// </summary>
-        /// <param name="attribute">A <see cref="IReadOnlyTagHelperAttribute"/> whose values should be copied.</param>
-        public TagHelperAttribute(IReadOnlyTagHelperAttribute attribute)
-            : this(attribute?.Name, attribute?.Value)
+        /// <param name="name">The <see cref="Name"/> of the attribute.</param>
+        /// <param name="value">The <see cref="Value"/> of the attribute.</param>
+        public TagHelperAttribute(string name, object value)
+            : this(name, value, minimized: false)
         {
-            if (attribute == null)
-            {
-                throw new ArgumentNullException(nameof(attribute));
-            }
-
-            Minimized = attribute.Minimized;
         }
 
         /// <summary>
-        /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>
-        /// and <paramref name="value"/>.
+        /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>,
+        /// <paramref name="value"/> and <paramref name="minimized"/>.
         /// </summary>
         /// <param name="name">The <see cref="Name"/> of the attribute.</param>
         /// <param name="value">The <see cref="Value"/> of the attribute.</param>
-        public TagHelperAttribute(string name, object value)
+        /// <param name="minimized">The <see cref="Minimized"/> value of the attribute.</param>
+        public TagHelperAttribute(string name, object value, bool minimized)
         {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
             Name = name;
             Value = value;
+            Minimized = minimized;
         }
 
         /// <summary>
         /// Gets or sets the name of the attribute.
         /// </summary>
-        public string Name { get; set; }
+        public string Name { get; }
 
         /// <summary>
         /// Gets or sets the value of the attribute.
         /// </summary>
-        public object Value { get; set; }
+        public object Value { get; }
 
         /// <summary>
         /// Gets or sets an indication whether the attribute is minimized or not.
```

Comment:
After reading @rynowak's [comment](https://github.com/aspnet/Razor/pull/682/files#r51336706), see this comment goes down the rabbit hole without saying much.